### PR TITLE
Add #include <limits>

### PR DESCRIPTION
**Pull request by lxmota**
Otherwise there are compile errors with gcc 11.1.1

### CODE REVIEW

**Review by nmm0 — APPROVED** [review]
Looks good, though I haven't set up a gcc 11 install yet. The tests failing seems to be an issue with the ubuntu image, not with the PR

**Comment by lxmota** [comment]
@nmm0 Thanks. Yes, the failing tests have nothing to do with the PR. I'm assuming the tests were set up by @uhetmaniuk
I'll wait until they are fixed to merge, just in case.

**Comment by lxmota** [comment]
Re-ran tests manually and successfully, merging now.

**Pull request merged**
